### PR TITLE
build(refs DPLAN-12315): Remove $on since its removed in vue3

### PR DESCRIPTION
**Pull request by salisdemos**
### Ticket
DPLAN-12315

don't use the $on$off events within the code-Section anymore. Its removed in vue3. To maintain the functionality, I used the events straight from the child-component.

### CODE REVIEW

File: client/js/components/procedure/Shared/SimilarStatementSubmitters/SimilarStatementSubmitters.vue [hunk 1]
```diff
@@ -309,6 +311,27 @@ export default {
...
+          type: 'SimilarStatementSubmitter'
+        }
+      })
+
+      this.listEntries.splice(index, 1)
+
+      if (this.isRequestFormPost === false) {
+        this.deleteSimilarStatementSubmitter()
+      }
+      if (this.isRequestFormPost) {
```
Comment (hwiem) [inline client/js/components/procedure/Shared/SimilarStatementSubmitters/SimilarStatementSubmitters.vue:330]: nitpick: add empty line between ifs

```suggestion
      
      if (this.isRequestFormPost) {
```

File: client/js/components/procedure/Shared/SimilarStatementSubmitters/SimilarStatementSubmitters.vue [hunk 2]
```diff
@@ -402,6 +425,15 @@ export default {
       }
     },
 
+    showUpdateForm (index) {
```
Comment (hwiem) [inline client/js/components/procedure/Shared/SimilarStatementSubmitters/SimilarStatementSubmitters.vue:428]: nitpick: I think the function name is not very descriptive; I'd suggest to rename it to something like `setInitialValues` or `setFormFields`

File: client/js/components/procedure/Shared/SimilarStatementSubmitters/SimilarStatementSubmitters.vue [hunk 3]
```diff
@@ -402,6 +425,15 @@ export default {
...
+      this.formFields.submitterCity = this.listEntries[index].submitterCity
+      this.formFields.submitterName = this.listEntries[index].submitterName
+      this.formFields.submitterAddress = this.listEntries[index].submitterAddress
+      this.formFields.submitterHouseNumber = this.listEntries[index].submitterHouseNumber
+      this.formFields.submitterPostalCode = this.listEntries[index].submitterPostalCode
+      this.formFields.submitterEmailAddress = this.listEntries[index].submitterEmailAddress
```
Comment (hwiem) [inline client/js/components/procedure/Shared/SimilarStatementSubmitters/SimilarStatementSubmitters.vue:434]: nitpick: I find this a little less verbose

```suggestion
      const {
        submitterAddress,
        submitterCity,
        submitterEmailAddress,
        submitterHouseNumber,
        submitterName,
        submitterPostalCode
      } = this.listEntries[index]
      
      this.formFields = {
        ...this.formFields, // not sure if this is needed
        submitterAddress,
        submitterCity,
        submitterEmailAddress,
        submitterHouseNumber,
        submitterName,
        submitterPostalCode
      }
```

File: client/js/components/procedure/basicSettings/DpEmailList.vue [hunk 4]
```diff
@@ -142,25 +145,18 @@ export default {
       this.$emit('saved', extraEmailAddress)
     },
 
+    showUpdateForm (index) {
+      this.formFields.mail = this.emails[index].mail
+      this.itemIndex = index
+    },
+
     updateExtraEmailAddress (index, extraEmailAddress) {
```
Comment (hwiem) [inline client/js/components/procedure/basicSettings/DpEmailList.vue:153]: Is this event handled anywhere?

File: client/js/components/procedure/basicSettings/DpEmailList.vue [hunk 5]
```diff
@@ -106,17 +108,18 @@ export default {
...
-      this.updateExtraEmailAddress(index)
-    },
-
     addElement () {
       this.emails.push({
         mail: this.formFields.mail
       })
     },
 
+    deleteEntry (index) {
```
Comment (hwiem) [inline client/js/components/procedure/basicSettings/DpEmailList.vue:117]: nitpick: In order to have one function do one thing, I'd rename this to `handleDelete`, move `this.emails.splice(index, 1)` (back) to a `delete` function and call it here. I'm not sure if `this.updateExtraEmailAddress(index)` is still needed?

File: client/js/components/statement/voter/StatementVoter.vue [hunk 6]
```diff
@@ -378,6 +379,12 @@ export default {
       return isEmpty
     },
 
+    deleteEntry (index) {
```
Comment (hwiem) [inline client/js/components/statement/voter/StatementVoter.vue:382]: nitpick: I'd call this `handleDelete`, but maybe it's a matter of personal preference

File: client/js/components/statement/voter/StatementVoter.vue [hunk 7]
```diff
@@ -392,6 +399,13 @@ export default {
         active: true
       }
       this.updating = false
+    },
+
+    showUpdateForm (index) {
+      for (const key in this.formFields) {
+        this.formFields[key] = this.getVoters[index][key]
+      }
```
Comment (hwiem) [inline client/js/components/statement/voter/StatementVoter.vue:407]: nitpick: I'd move this to its own function, e.g. `setFormFields`, call it here and add an empty line below

File: client/js/components/user/CustomerSettings/CustomerSettingsSupport.vue [hunk 8]
```diff
@@ -201,6 +209,13 @@ export default {
       this.updating = false
     },
 
+    showUpdateForm (index) {
+      this.updateForm(index)
+      this.$nextTick(() => {
+        document.getElementById('contactForm').scrollIntoView()
+      })
```
Comment (hwiem) [inline client/js/components/user/CustomerSettings/CustomerSettingsSupport.vue:216]: nitpick

```suggestion
      this.updateForm(index)
      
      this.$nextTick(() => {
        document.getElementById('contactForm').scrollIntoView()
      })
```

File: package.json [hunk 9]
```diff
@@ -186,7 +186,9 @@
     "ff esr"
   ],
   "resolutions": {
-    "@mapbox/mapbox-gl-style-spec": "13.23.1"
+    "@mapbox/mapbox-gl-style-spec": "13.23.1",
```
Comment (hwiem) [inline package.json:189]: Should these changes be here?

**Review by hwiem — CHANGES_REQUESTED** [review]
I'm not sure about the package.json changes; apart from that only nitpicks

File: client/js/components/procedure/Shared/SimilarStatementSubmitters/SimilarStatementSubmitters.vue (hunk 1 above)
Comment (salisdemos) [reply client/js/components/procedure/Shared/SimilarStatementSubmitters/SimilarStatementSubmitters.vue:331]: e13ad3e588c43000eb5bbcbd90664bc9919e7c04

File: client/js/components/procedure/basicSettings/DpEmailList.vue (hunk 5 above)
Comment (salisdemos) [reply client/js/components/procedure/basicSettings/DpEmailList.vue:117]: like so?
e13ad3e588c43000eb5bbcbd90664bc9919e7c04

File: client/js/components/statement/voter/StatementVoter.vue (hunk 7 above)
Comment (salisdemos) [reply client/js/components/statement/voter/StatementVoter.vue:419]: I think thats a bit too much. I left it like it is.

File: client/js/components/procedure/Shared/SimilarStatementSubmitters/SimilarStatementSubmitters.vue (hunk 3 above)
Comment (salisdemos) [reply client/js/components/procedure/Shared/SimilarStatementSubmitters/SimilarStatementSubmitters.vue]: I found a even shorter form ;-) 
e13ad3e588c43000eb5bbcbd90664bc9919e7c04

File: client/js/components/procedure/basicSettings/DpEmailList.vue (hunk 4 above)
Comment (salisdemos) [reply client/js/components/procedure/basicSettings/DpEmailList.vue]: nope e13ad3e588c43000eb5bbcbd90664bc9919e7c04

File: client/js/components/statement/voter/StatementVoter.vue (hunk 6 above)
Comment (salisdemos) [reply client/js/components/statement/voter/StatementVoter.vue]: why not. e13ad3e588c43000eb5bbcbd90664bc9919e7c04

**Review by hwiem — APPROVED** [review]
lgtm, thanks! (code review only)

**Pull request merged**